x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java

```diff
@@ -594,6 +663,21 @@ void incrementDifferentiatorAndCreateNewProfile(Subject subject, ProfileDocument
         createNewProfile(subject, newUid, listener);
     }
 
+    private DomainConfig getDomainConfigForSubject(Subject subject) {
+        final RealmDomain realmDomain = subject.getRealm().getDomain();
+        if (realmDomain != null) {
+            final DomainConfig domainConfig = domainConfigLookup.apply(realmDomain.name());
+            if (domainConfig == null) {
+                throw new ElasticsearchException(
+                    "subject realm is under a domain [" + realmDomain.name() + "], but no associated domain config is found"
+                );
+            }
```

Comment:
Many adjustments to the tests are due to this check here. Previously it is OK to call the methods with arbitrary authentication object with domain. But it is no longer possible because this piece code checks that if the authentication has a domain, the domain must be configured in the settings.